docker/tarfile/src.go

```diff
@@ -180,10 +180,12 @@ func (s *Source) loadTarManifest() (*manifestItem, error) {
 	if err := json.Unmarshal(bytes, &items); err != nil {
 		return nil, errors.Wrap(err, "Error decoding tar manifest.json")
 	}
-	if len(items) != 1 {
-		return nil, errors.Errorf("Unexpected tar manifest.json: expected 1 item, got %d", len(items))
```

Comment:
This check should remain in place when using `Source` as a `types.ImageSource` backend.